### DISCUSSION TITLE
Allow to highlight lines in a code block (github flavored markdown)

### DISCUSSION THREAD

**Discussion by muescha** [community/community#42489] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

# where:
- all places where i can use [GitHub Flavored Markdown](https://github.github.com/gfm/#fenced-code-blocks) - issues, pull requests, wiki, comments and more ...

# what i like:
I have a fenced code blocks. I like to mark some lines as highlighted. 

To highlight lines, you can use one of the following directives as comments in your code:

- `highlight-line` highlights the current line;
- `highlight-next-line` highlights the next line;
- `highlight-start` highlights the lines until the matching hightlight-end;
- `highlight-range{1, 4-6}` will highlight the next line, and the fourth, fifth and sixth lines.

since detecting comments of a specific language and removing code i would not prefer i suggest a marker at the start of the codeblock would be also enough for this line highlighting:

````
```javascript{2,4-6}
````

or better as a named property 

````
```javascript:highlight={2,4-6}
````

named properties are better so that this is open for other properties like `title=`

````
```javascript:highlight={2,4-6}:title="Setup: ~/user/config.js"
````
 
# existing implementations and examples

see: https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-remark-prismjs#line-highlighting

# Note:
with highlight I don't mean [syntax highlighting](highlighting) for a specific language. this is still available as:

````
```javascript
var a = "b";
```
````

**Comment by bartvdbraak** [comment]
No updates or mentions of this feature? 😢

**Comment by adamoceallaigh** [comment]
As a workaround, you can utilise Github flavoured Markdown diff attribute. 

So, instead of specifying a language after your 3 backticks, you could specify diff.
Furthermore, add a '+' to any of the lines you want highlighted in green, '-' to any of the lines highlighted in red. 
So, the two options for highlighting are currently green and red, but it's better than no solution i guess

Example:

```diff
+ Chelsea are going to be Premier League Champions 2023/2024
- Arsenal are going to be Premier League Champions 2023/2024
```

**Reply by vincentKool** [reply]
Thanks for this workaround.

**Comment by MarioVilas** [comment]
Highlighting lines is fine, but it could also be interesting to add some way to highlight arbitrary chunks of text, for example by using a single backtick character. This might make things slightly incompatible with vanilla Markdown, since there's supposed to be nothing inside a code block that can be parsed as markup - but I suppose that can be mitigated by using a directive at the beginning, and it would not break existing parsers, it would just cause these backtick characters to be displayed.

This is what I mean:
````
```highlight
this is some text
this text is `highlighted' somewhere
this text is \`not highlighted\' because the backticks are escaped
```
````

Does this make sense?

**Reply by theswerd** [reply]
For my use case this doesn't. I'd like JS Syntax highlighting, combined with the ability to highlight specific sections. However, I see the value in this for other use cases.

**Comment by joshdunnlime** [comment]
This would be nice. What would also be nice is being able to make characters **bold** or _italicised_ in code blocks to emphasis small changes in bigger blocks:

```python
foo = bar(**_baz_**)  # doesn't work
```

It works in quote blocks

> foo = bar(**_baz_**)  # works in quotes but no syntax highlighting so harder to see

**Comment by meduzen** [comment]
To add to the existing implementations: [Vitepress one](https://vitepress.dev/guide/markdown#line-highlighting-in-code-blocks).

**Comment by calishu** [comment]
Nearly 3 years ago and still no answer from GitHub.

**Comment by Crissov** [comment]
The place for discussing extensions to Commonmark, the base of GFM, also has a [thread](https://talk.commonmark.org/t/code-blocks-extension-for-filenames-and-line-numbers/536) on extending the info string of fenced code blocks to support line numbering etc. (and several related ones). 

Although extensions will not be added to CM before it reaches v1.0 (if ever), this still seems like a better way forward than hoping for Github to add this unilaterally.

**Comment by danirui** [comment]
See also my feature request here https://github.com/orgs/community/discussions/153645 and a very small amount of updates/news from GitHub
